src/arrayMethodReduce.js

```diff
@@ -5,8 +5,22 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue;
+
+    if (arguments.length === 1) {
+      accumulator = this[0];
+
+      for (let i = 1; i < this.length; i++) {
+        accumulator = callback(accumulator, this[i], i, this);
+      }
+    } else {
+      for (let i = 0; i < this.length; i++) {
+        accumulator = callback(accumulator, this[i], i, this);
+      }
```

Comment:
You repeat the same logic in 2 cases. You can get rid of if/else statement by using a ternary operator.
```
const accumulator = ... ? ... : ... ;
const index = ... ? ... : ... ;

for (let i = index .....